### ISSUE TITLE
WebSocket autoFragment does not fragment based on maxFrameSize

### ISSUE THREAD

**Issue by lachlan-roberts** [eclipse/jetty.project#4152]
WebSocket auto fragmentation is defined in javadoc as `If true, frames are automatically fragmented to respect the maximum frame size.`  However the implementation in `Parser` does not implement this definition of auto fragment.

In `Parser` frames are auto fragmented if auto fragment is set to true, it is a data frame, and we don't currently have enough data to finish the frame. It does not fragment frames based on the `maxFrameSize`.

We have the following check in `Parser` which will not verify the `maxFrameSize` if auto fragment is set. So this check is assuming that auto fragment will fragment the frames to the max frame size.
```java
long maxFrameSize = configuration.getMaxFrameSize();
if (!configuration.isAutoFragment() && maxFrameSize > 0 && payloadLength > maxFrameSize)
    throw new MessageTooLargeException("Cannot handle payload lengths larger than " + maxFrameSize);
```

Either the javadoc definition of auto fragment needs to be changed, or the implementation in `Parser` should be changed to enforce frames are fragmented to the `maxFrameSize`.

**Comment by lachlan-roberts** [comment]
Now that PRs #4219, #4260 and #4232 are merged autoFragment is working as intended.

**Comment by joakime** [comment]
@lachlan-roberts a little github PR magic trick.

If you include in the original PR message the issues that this PR belongs to, with the keyword "Closes", each on their own line, at the end of the message, like this ....

> Closes #4219 
> Closes #4260
> Closes #4232

Then when you merge the PR the issue automatically closes for you.
